src/cpp/astar.cpp

```diff
@@ -24,18 +24,60 @@ bool operator<(const Node &n1, const Node &n2) {
   return n1.cost > n2.cost;
 }
 
+enum Heuristic { DEFAULT, DIAGONAL_DISTANCE, MANHATTAN_DISTANCE, ORTHOGONAL_X, ORTHOGONAL_Y };
+
 // See for various grid heuristics:
 // http://theory.stanford.edu/~amitp/GameProgramming/Heuristics.html#S7
 // L_\inf norm (diagonal distance)
-inline float linf_norm(int i0, int j0, int i1, int j1) {
+inline float linf_norm(int i0, int j0, int i1, int j1, int, int) {
   return std::max(std::abs(i0 - i1), std::abs(j0 - j1));
 }
 
 // L_1 norm (manhattan distance)
-inline float l1_norm(int i0, int j0, int i1, int j1) {
+inline float l1_norm(int i0, int j0, int i1, int j1, int, int) {
   return std::abs(i0 - i1) + std::abs(j0 - j1);
 }
 
+// Orthogonal x (moves by x first, then half way by y)
+inline float orthogonal_x(int i0, int j0, int i1, int j1, int i2, int j2) {
+  int di = std::abs(i0 - i1);
+  int dim = std::abs(i1 - i2);
+  int djm = std::abs(j1 - j2);
+  if (di > (dim * 0.5)) {
+    return di + djm;
+  } else {
+    return std::abs(j0 - j1);
+  }
+}
+
+// Orthogonal y (moves by y first, then half way by x)
+inline float orthogonal_y(int i0, int j0, int i1, int j1, int i2, int j2) {
+  int dj = std::abs(j0 - j1);
+  int djm = std::abs(j1 - j2);
+  int dim = std::abs(i1 - i2);
+  if (dj > (djm * 0.5)) {
+    return dj + dim;
+  } else {
+    return std::abs(i0 - i1);
+  }
+}
```

Comment:
Let's add a note that these are experimental and the caller accepts a performance penalty for prettier paths (if I understand their purpose correctly).

We could even consider moving them to another file to make this explicit.

Reply:
Your right, will move to another file pretty_heuristics.cpp